No default encoding kinesis

danamansana opened the pull request:
Also some changes to the gemspec that were never versioned.

danamansana committed:
Update gem

danamansana committed:
Update to 0.0.1

danamansana committed:
Build 0.0.1

danamansana committed:
Add dependencies to gemspec

danamansana committed:
Fix typo

danamansana committed:
Remove base64 encoding on write in kinesis client

mwbenowitz reviewed (approved):
👍 LGTM

mwbenowitz commented:
@danamansana found a fun unanticipated side effect! If we don't base64 encode the avro record it is a string of byte codes, and the logger can't handle that because it expects `utf-8` and the kinesis client throws an error. I'm trying to think of what the cleanest way to handle this would be, will let you know if I come up with something

mwbenowitz committed:
NOREF Output UTF-8 string from Avro encoding

When not base64 encoding Avro records, the method should return a UTF-8 encoded string. Otherwise other modules such as logging and JSON cannot handle the ASCII string that is usually returned.

This enforces this by setting the encoding on the StringIO object that generates the returned string from the encode method. This should make it safe to use in all contexts.

If base64 encoding the result the string is still in ASCII, though this is safe as a base64 string can be trusted to contain only well-formed/parseable characters

mwbenowitz committed:
Merge pull request #4 from NYPL/avro-encoding-to-utf8

NOREF Output UTF-8 string from Avro encoding
